shoop/campaigns/models/campaign_effects.py

```diff
@@ -0,0 +1,110 @@
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from enum import Enum
+
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from polymorphic.models import PolymorphicModel
+
+from shoop.admin.forms.fields import PercentageField
+from shoop.core.fields import MoneyValueField
+from shoop.utils.properties import PriceProperty
+
+
+class EffectType(Enum):
+    CATALOG = 1
+    BASKET = 2
+
+
+class CampaignEffect(PolymorphicModel):
+    identifier = None
+    model = None
+    admin_form_class = None
+    yields = False
+
+    def set_locals(self, **kwargs):
+        self.order_source = kwargs.get("order_source")
+        self.price_info = kwargs.get("price_info")
+        self.context = kwargs.get("context")
+        self.product = kwargs.get("product")
+        self._campaign = kwargs.get("campaign")
+        self.set_type()
```

Comment:
These shouldn't be here in CampaignEffect but rather in a separate class like `CampaignApplyingContext`. Then instance of that class should be passed to `apply`.

Also: Describe what are `price_info`, `context` and `product` in that list. I don't see which price, what context (pricing context? something else?) and which product they are referring to.